doc/source/tune/api_docs/checkpointing/class-checkpointing.rst

```diff
@@ -0,0 +1,24 @@
+Class API Checkpointing
+~~~~~~~~~~~~~~~~~~~~~~~
+
+You can also implement checkpoint/restore using the Trainable Class API:
+
+.. code-block:: python
+
+    class MyTrainableClass(Trainable):
```

Comment:
@justinvyu I realise you're just moving this from another spot. But since we're already touching it, can we please move the code into a file (in the `doc_code` folder) and `literalinclude` it from there?

For instance, here we're missing all imports, and we want runnable snippets.